hadoop-hdds/docs/content/feature/Snapshot.md

```diff
@@ -23,55 +23,260 @@ summary: Ozone Snapshot
+* **Access and Restore:** Reading from an Ozone snapshot is done through the special `.snapshot` path in the bucket, analogous to HDFS where snapshots appear in a `.snapshot` directory under the snapshottable directory. Restoring data from a snapshot in Ozone is a manual process of copying data out of the `.snapshot` directory back into the active area if needed (or using DistCp for large-scale copy). There is no automatic “rollback” of a bucket to a snapshot, since Ozone treats snapshots as read-only backups (similar approach in HDFS – HDFS snapshots can be used to restore files but not instant revert of the live state without manual intervention).
+* **Implementation:** Internally, Ozone snapshots leverage the OM’s key-value store and share data blocks between snapshots, which is analogous to HDFS’s copy-on-write approach for metadata. One notable difference is that Ozone being object-store, does not have an equivalent of HDFS block tracking for snapshots at the DataNode level – all the intelligence is in the OM. HDFS NameNode also handles snapshot metadata, but Ozone’s design (with its RocksDB metadata store) means snapshot creation is O(1) and implemented via metadata pointers rather than duplication. Both systems do not copy actual file data on snapshot – they rely on not deleting underlying blocks.
+
+In summary, Ozone snapshots give similar benefits for Ozone buckets as HDFS snapshots do for directories, but Ozone’s approach is bucket-scoped and oriented toward object store semantics. If coming from HDFS, just note that you cannot snapshot volumes or the entire namespace at once (you snapshot each bucket individually), and you cannot snapshot sub-folders of a bucket.
+
+## **Known Issues and Limitations**
+
+While Apache Ozone’s snapshot feature is powerful, there are a few current limitations and caveats to be aware of:
+
+* **S3 Interface Support:** Snapshot operations are *not* supported via the S3 API (`s3://` or `s3a://` access). Ozone’s S3 Gateway does not expose snapshot functionality, since Amazon S3 has no direct concept of snapshots. This means you cannot create or list snapshots through the S3 Gateway or the `s3a` Hadoop FileSystem connector. Snapshots must be managed via the Ozone RPC interface (ozone shell, ozone fs, or Ozone Client API). However, snapshotted data can be read via the S3 Gateway using the special `.snapshot` key prefix. For example, a key `key1` in the snapshot `snap1` can be read using the path `.snapshot/snap1/key1`.
```

Comment:
We support snapshots in S3 buckets via creating linked buckets on s3v volume

Reply:
Sure but S3 API does not have the notion of snapshots